onnxruntime/core/providers/dml/DmlExecutionProvider/src/MLOperatorAuthorImpl.cpp

```diff
@@ -1120,57 +1120,63 @@
-            m_graphNodeCreateInfo->kernelOutputIndices.insert(
-                m_graphNodeCreateInfo->kernelOutputIndices.begin(),
-                dmlProperties->kernelOutputIndices,
-                dmlProperties->kernelOutputIndices + dmlProperties->dmlOutputCount);
-        }
-        else
-        {
-            m_graphNodeCreateInfo->kernelOutputIndices.resize(dmlProperties ? dmlProperties->dmlOutputCount : GetOutputCount());
-            std::iota(m_graphNodeCreateInfo->kernelOutputIndices.begin(), m_graphNodeCreateInfo->kernelOutputIndices.end(), 0);
-        }
-
-        m_graphNodeCreateInfo->allowHalfPrecisionComputation = dmlProperties ? dmlProperties->allowHalfPrecisionComputation : true;
-    }
-
     HRESULT STDMETHODCALLTYPE DmlGraphOpKernelInfoWrapper::SetDmlOperator(
-        IDMLOperator* op,
-        _In_ const DML_OPERATOR_DESC* desc,
-        _In_opt_ const MLOperatorKernelDmlProperties* dmlProperties) const noexcept
+        _In_ const MLOperatorGraphDesc* operatorGraphDesc) const noexcept
     {
         ORT_TRY
         {
-          ML_CHECK_BOOL(op != nullptr);
-          ML_CHECK_BOOL(dmlProperties != nullptr);
+            ML_CHECK_BOOL_MSG(operatorGraphDesc != nullptr, "OperatorGraphDesc can't be null while creating operator kernel for DML Graph");
+            // Either nodesAsOpDesc or nodesIDMLOperator can be present.
+            assert(operatorGraphDesc->nodeCount == 0 || (!operatorGraphDesc->nodesAsOpDesc ^ !operatorGraphDesc->nodesAsIDMLOperator));
+
+            if (operatorGraphDesc->nodesAsOpDesc)
+            {
+                m_graphNodeCreateInfo->nodesAsOperatorDesc = std::vector<std::unique_ptr<AbstractOperatorDesc>>();
+                for (uint32_t nodeIndex = 0; nodeIndex < operatorGraphDesc->nodeCount; nodeIndex++) 
+                {
+                    auto* node = operatorGraphDesc->nodesAsOpDesc[nodeIndex];
+                    assert(node != nullptr);
+                    AbstractOperatorDesc abstractDesc = SchemaHelpers::ConvertOperatorDesc(*node);
+                    m_graphNodeCreateInfo->nodesAsOperatorDesc.push_back(std::make_unique<AbstractOperatorDesc>(std::move(abstractDesc)));
+                }
+            }
+            else
+            {
+                m_graphNodeCreateInfo->nodesAsIDMLOperator = std::vector<Microsoft::WRL::ComPtr<IDMLOperator>>();
+                for (uint32_t nodeIndex = 0; nodeIndex < operatorGraphDesc->nodeCount; nodeIndex++) 
+                {
+                    auto* node = operatorGraphDesc->nodesAsIDMLOperator[nodeIndex];
+                    assert(node != nullptr);
+                    m_graphNodeCreateInfo->nodesAsIDMLOperator.push_back(node);
+                }
+            }
 
-          m_graphNodeCreateInfo->initialized = true;
+            // Need to iterate on each edges to check for nullptr
+            assert(operatorGraphDesc->nodeCount == 0 || (operatorGraphDesc->inputEdgeCount > 0 && operatorGraphDesc->inputEdges != nullptr));
```

Comment:
>operatorGraphDesc->nodeCount == 0 || ( [](http://example.com/codeflow?start=19&length=38)

Is it even possible for node count to be zero and inputEdgeCount to be nonzero? Even if it possible, the nullability of `inputEdges` should depend solely on its counterpart contract field `inputEdgeCount`, not nodeCount.